pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandlerTest.java

```diff
@@ -460,11 +538,125 @@ public void testComputeOperation()
     assertEquals(operationMap.get(config1.getName()), ForwardIndexHandler.Operation.CHANGE_RAW_INDEX_COMPRESSION_TYPE);
     assertEquals(operationMap.get(config2.getName()), ForwardIndexHandler.Operation.CHANGE_RAW_INDEX_COMPRESSION_TYPE);
 
+    // TEST10: Disable forward index for a column which already has forward index disabled
+    indexLoadingConfig = new IndexLoadingConfig(null, _tableConfig);
+    indexLoadingConfig.getForwardIndexDisabledColumns().add(DIM_SV_FORWARD_INDEX_DISABLED_INTEGER);
+    fwdIndexHandler = new ForwardIndexHandler(existingSegmentMetadata, indexLoadingConfig, null);
+    operationMap = fwdIndexHandler.computeOperation(writer);
+    assertEquals(operationMap, Collections.EMPTY_MAP);
+
+    // TEST11: Disable forward index for a dictionary column with forward index enabled
+    indexLoadingConfig = new IndexLoadingConfig(null, _tableConfig);
+    indexLoadingConfig.getForwardIndexDisabledColumns().add(DIM_DICT_INTEGER);
+    indexLoadingConfig.getInvertedIndexColumns().add(DIM_DICT_INTEGER);
+    fwdIndexHandler = new ForwardIndexHandler(existingSegmentMetadata, indexLoadingConfig, _schema);
+    operationMap = fwdIndexHandler.computeOperation(writer);
+    assertEquals(operationMap.size(), 1);
+    assertEquals(operationMap.get(DIM_DICT_INTEGER), ForwardIndexHandler.Operation.DELETE_FORWARD_INDEX);
+
+    // TEST12: Disable forward index for a dictionary column with forward index enabled, disable dictionary
+    indexLoadingConfig = new IndexLoadingConfig(null, _tableConfig);
+    indexLoadingConfig.getForwardIndexDisabledColumns().add(DIM_LZ4_BYTES);
+    indexLoadingConfig.getNoDictionaryColumns().add(DIM_LZ4_BYTES);
+    indexLoadingConfig.getInvertedIndexColumns().add(DIM_LZ4_BYTES);
+    fwdIndexHandler = new ForwardIndexHandler(existingSegmentMetadata, indexLoadingConfig, _schema);
+    ForwardIndexHandler finalFwdIndexHandler = fwdIndexHandler;
+    assertThrows(IllegalStateException.class, () -> finalFwdIndexHandler.computeOperation(writer));
+
+    // TEST13: Disable forward index for a raw column with forward index enabled and enable dictionary
```

Comment:
(nit): enable dictionary and inverted index.

Reply:
done